Source/Search/Mixins/AdvancedSearchProviderMixin.lua

```diff
@@ -0,0 +1,143 @@
+local _, addonTable = ...
+local zc = addonTable.zc
+
+AuctionatorAdvancedSearchProviderMixin = CreateFromMixins(AuctionatorMultiSearchMixin, AuctionatorSearchProviderMixin)
+
+local KEYWORD_SEARCH_EVENTS = {
+  "AUCTION_HOUSE_BROWSE_RESULTS_UPDATED",
+  "AUCTION_HOUSE_BROWSE_RESULTS_ADDED",
+  "AUCTION_HOUSE_BROWSE_FAILURE"
+}
+
+local function IsCompoundSearch (searchString)
+  if searchString == nil then
```

Comment:
```lua
return 
  searchString ~= nil and 
  Auctionator.Utilities.StringContains(searchString, Auctionator.Constants.AdvancedSearchSeparator)
```

Reply:
Uh, that wasn't the problem...

Reply:
The comment is related to the logic, not whatever problem it was solving. I try to avoid constructs like:
```lua
if true then
  return false
else
  return boolean
```
This logic can be made more concise by simply returning the combined boolean expression.